app/models/message.rb

```diff
@@ -9,10 +9,18 @@ class Message
   validates :content, presence: true
 
   after_create :add_index
+  after_update :add_index
+  before_destroy :delete_index
```

Comment:
I heard that callbacks sometime do not work correctly in case of cassandra. Did you check it?